### PR TITLE
fix: [DHIS2-17131] schedule dates with format DD-MM-YYYY

### PR DESCRIPTION
**Pull request by superskip**
[DHIS2-17131](https://dhis2.atlassian.net/browse/DHIS2-17131)

The dates appears to already be on standard system format (yyyy-mm-dd) in `deriveAutoGenerateEvents()`.

[DHIS2-17131]: https://dhis2.atlassian.net/browse/DHIS2-17131?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/core_modules/capture-core/components/Pages/New/RegistrationDataEntry/helpers/deriveAutoGenerateEvents.js [hunk 1]
```diff
@@ -56,13 +53,13 @@ export const deriveAutoGenerateEvents = ({
                 const eventInfo = openAfterEnrollment
                     ? {
                         status: 'ACTIVE',
-                        occurredAt: convertFn(dateToUseInActiveStatus, dataElementTypes.DATE),
-                        scheduledAt: convertFn(dateToUseInActiveStatus, dataElementTypes.DATE),
+                        occurredAt: convertClientToServer(dateToUseInActiveStatus, dataElementTypes.DATE),
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/Pages/New/RegistrationDataEntry/helpers/deriveAutoGenerateEvents.js:56]: Hey,
Do we need to do any conversion here at all? By looking at [useBuildEnrollmentPayload](https://github.com/dhis2/capture-app/blob/master/src/core_modules/capture-core/components/DataEntries/EnrollmentRegistrationEntry/hooks/useBuildEnrollmentPayload.js#L94C43-L94C71), I can see that `enrolledAt` and `occurredAt` were already converted from [clientToServer](https://github.com/dhis2/capture-app/blob/master/src/core_modules/capture-core/components/DataEntries/EnrollmentRegistrationEntry/hooks/useBuildEnrollmentPayload.js#L61) and passed to deriveAutoGenerateEvents. The `dateToUseInActiveStatus` value can either be enrolledAt or occurredAt and they are in the server format already. WDYT?
Thanks!

Comment (JoakimSM) [reply src/core_modules/capture-core/components/Pages/New/RegistrationDataEntry/helpers/deriveAutoGenerateEvents.js:56]: Good catch @simonadomnisoru. Please remove the conversion here @superskip.

Comment (superskip) [reply src/core_modules/capture-core/components/Pages/New/RegistrationDataEntry/helpers/deriveAutoGenerateEvents.js:56]: Thanks, makes sense, will fix :+1:

**Review by simonadomnisoru — APPROVED** [review]

**Review by JoakimSM — APPROVED** [review]

**Review by geethaalwan — APPROVED** [review]
Tested successfully on 2.42,2.41.0,2.40.4,2.39.6,2.38.7 versions

**Pull request merged**